src/datasources/email/email.datasource.spec.ts

```diff
@@ -329,4 +329,74 @@ describe('Email Datasource Tests', () => {
       }),
     ).rejects.toThrow(EmailAddressDoesNotExistError);
   });
+
+  it('updates emails successfully', async () => {
+    const chainId = faker.number.int({ max: DB_CHAIN_ID_MAX_VALUE }).toString();
+    const safeAddress = faker.finance.ethereumAddress();
+    const prevEmailAddress = new EmailAddress(faker.internet.email());
+    const emailAddress = new EmailAddress(faker.internet.email());
+    const account = faker.finance.ethereumAddress();
+    const code = faker.string.numeric();
+    const codeGenerationDate = faker.date.recent();
+
+    await target.saveEmail({
+      chainId,
+      safeAddress,
+      emailAddress: prevEmailAddress,
+      account,
+      code: faker.string.numeric(),
+      codeGenerationDate: faker.date.recent(),
+    });
+
+    await target.verifyEmail({
+      chainId,
+      safeAddress,
+      account,
+    });
```

Comment:
Verification is optional when updating an email right? As in we can update unverified emails.

Reply:
I've added a second test in 5674ff88d2bdb104e784280af3968346b9446d87. The two check that updates are "correct" (setting the verified flag to `false`, etc) when the current email was (un-)verified.